### ISSUE TITLE
Problem with option grabbing argument value

### ISSUE THREAD

**Issue by AndyAyersMS** [dotnet/command-line-api#532]
I have `--platform` as a `OneOrMany` string option, and the command also has a `OneOrMany` argument. For some reason the platform option reaches over intervening options and grabs what is meant to be the argument value.
```
jit-dasm-pmi [parse] 
--platform D:\repos\coreclr\bin\tests\Windows_NT.x64.Release\Tests\Core_Root 
--verbose 
--corerun D:\repos\coreclr\bin\tests\Windows_NT.x64.Release\Tests\Core_Root\corerun.exe
--jit d:\repos\coreclr2\bin\Product\Windows_NT.x64.Checked\clrjit.dll 
--nocopy 
--output D:\repos\coreclr\bin\diffs\base
D:\repos\coreclr\bin\tests\Windows_NT.x64.Release\Tests\Core_Root\System.Private.CoreLib.dll

![ jit-dasm-pmi 
  [ --platform <D:\repos\coreclr\bin\tests\Windows_NT.x64.Release\Tests\Core_Root> <D:\repos\coreclr\bin\tests\Windows_NT.x64.Release\Tests\Core_Root\System.Private.CoreLib.dll> ] 
  [ --verbose <True> ] 
  [ --corerun <D:\repos\coreclr\bin\tests\Windows_NT.x64.Release\Tests\Core_Root\corerun.exe> ] 
  [ --jit <d:\repos\coreclr2\bin\Product\Windows_NT.x64.Checked\clrjit.dll> ] 
  [ --nocopy <True> ] 
  [ --output <D:\repos\coreclr\bin\diffs\base> ] ]
```

**Comment by jonsequitur** [comment]
The bug was likely introduced when #383 was resolved.

Related: #527.

**Comment by jonsequitur** [comment]
It appears that this was fixed with #527. @AndyAyersMS if you see it still happens, please let us know.
